test/controllers/tools_controller_test.rb

```diff
@@ -7,5 +7,6 @@ class ToolsControllerTest < ActionDispatch::IntegrationTest
   test "should get index" do
     get tools_url
     assert_response :success
+    assert_select '.featured-article'
```

Comment:
Recommend to not test rendering of CSS classes in controller test. This can be removed.

Reply:
The goal here was more to assert that a featured article is rendered.
In general what's your test strategy. Looks like most of the model tests are empty and apparently there isn't much in the controller tests either.

Reply:
Test strategy is to make sure coverage is maintained for new controller actions, to test conditional branches that may occur, but to do more surface testing. Controller tests should mostly look for redirects on failure, or success of page rendering, as opposed to what the contents of the views ends up being. Models are tested if model code isn't reached via a controller test, or if there is a more complex public interface for a model.

Reply:
Additionally, featured models don't necessarily render on each page, following pages (page 2) would not show featured articles, as only the top three in the scope would appear on page 1.

Reply:
If that's the desired behavior I'll change the query.

Reply:
Testing views is done in system tests using rails test:system, as it fits more in the purpose of navigating between pages, clicking buttons in the view, etc.